src/connection/mc_worker.erl

```diff
@@ -50,19 +51,30 @@ init(Options) ->
       Login = mc_utils:get_value(login, Options),
       Password = mc_utils:get_value(password, Options),
       NextReqFun = mc_utils:get_value(next_req_fun, Options, fun() -> ok end),
-      case auth_if_credentials(Socket, ConnState, NetModule, Login, Password) of
+      case mc_worker_pid_info:install_mc_worker_info(Socket,
+                                                     NetModule,
+                                                     ConnState#conn_state.database) of
         ok ->
-          proc_lib:init_ack({ok, self()}),
-          gen_server:enter_loop(?MODULE, [],
-              #state{socket = Socket,
-                     conn_state = ConnState,
-                     net_module = NetModule,
-                     next_req_fun = NextReqFun});
-        {error, Reason} ->
-          proc_lib:init_ack({error, Reason})
+          case auth_if_credentials(Socket, ConnState, NetModule, Login, Password) of
+            ok ->
+              proc_lib:init_ack({ok, self()}),
+              gen_server:enter_loop(?MODULE, [],
+                  #state{socket = Socket,
+                         conn_state = ConnState,
+                         net_module = NetModule,
+                         next_req_fun = NextReqFun}),
+              ignore;
+            {error, Reason} ->
+              proc_lib:init_ack({error, Reason}),
+              ignore
+          end;
+        Error ->
+          proc_lib:init_ack(Error),
```

Comment:
Tiny nit: maybe just add a comment saying that `Socket` gets closed in `install_mc_worker_info` when it fails, for future reference.

```suggestion
        Error ->
          %% Note: the socket is closed by `install_mc_worker_info', so it won't leak here.
          proc_lib:init_ack(Error),
```